server/enterprise/message_export/csv_export/csv_export.go

```diff
@@ -250,84 +282,78 @@ func mergePosts(left []*model.MessageExport, right []*model.MessageExport) func(
+	var joinLeavePosts []*model.MessageExport
 
-		joins, leaves := shared.GetJoinsAndLeavesForChannel(channel.StartTime, channel.EndTime, channelMembersHistory, postAuthorsByChannel[channel.ChannelId])
-
-		for _, join := range joins {
-			enterMessage := fmt.Sprintf("User %s (%s) joined the channel", join.Username, join.Email)
-			enterPostType := EnterPostType
-			createAt := model.NewPointer(join.Datetime)
-			channelCopy := channel
-			if join.Datetime <= channel.StartTime {
-				enterPostType = PreviouslyJoinedPostType
-				enterMessage = fmt.Sprintf("User %s (%s) was already in the channel", join.Username, join.Email)
-				createAt = model.NewPointer(channel.StartTime)
-			}
-			joinLeavePosts = append(
-				joinLeavePosts,
-				&model.MessageExport{
-					TeamId:          channel.TeamId,
-					TeamName:        channel.TeamName,
-					TeamDisplayName: channel.TeamDisplayName,
-
-					ChannelId:          &channelCopy.ChannelId,
-					ChannelName:        &channelCopy.ChannelName,
-					ChannelDisplayName: &channelCopy.ChannelDisplayName,
-					ChannelType:        &channelCopy.ChannelType,
-
-					UserId:    model.NewPointer(join.UserId),
-					UserEmail: model.NewPointer(join.Email),
-					Username:  model.NewPointer(join.Username),
-					IsBot:     join.IsBot,
-
-					PostId:         model.NewPointer(""),
-					PostCreateAt:   createAt,
-					PostMessage:    &enterMessage,
-					PostType:       &enterPostType,
-					PostOriginalId: model.NewPointer(""),
-					PostFileIds:    []string{},
-				},
-			)
-		}
-		for _, leave := range leaves {
-			leaveMessage := fmt.Sprintf("User %s (%s) leaved the channel", leave.Username, leave.Email)
-			leavePostType := LeavePostType
-			channelCopy := channel
-
-			joinLeavePosts = append(
-				joinLeavePosts,
-				&model.MessageExport{
-					TeamId:          channel.TeamId,
-					TeamName:        channel.TeamName,
-					TeamDisplayName: channel.TeamDisplayName,
-
-					ChannelId:          &channelCopy.ChannelId,
-					ChannelName:        &channelCopy.ChannelName,
-					ChannelDisplayName: &channelCopy.ChannelDisplayName,
-					ChannelType:        &channelCopy.ChannelType,
-
-					UserId:    model.NewPointer(leave.UserId),
-					UserEmail: model.NewPointer(leave.Email),
-					Username:  model.NewPointer(leave.Username),
-					IsBot:     leave.IsBot,
-
-					PostId:         model.NewPointer(""),
-					PostCreateAt:   model.NewPointer(leave.Datetime),
-					PostMessage:    &leaveMessage,
-					PostType:       &leavePostType,
-					PostOriginalId: model.NewPointer(""),
-					PostFileIds:    []string{},
-				},
-			)
+	joins, leaves := shared.GetJoinsAndLeavesForChannel(startTime, endTime, channelMembersHistory, postAuthors)
+
+	for _, join := range joins {
+		enterMessage := fmt.Sprintf("User %s (%s) joined the channel", join.Username, join.Email)
+		enterPostType := EnterPostType
+		createAt := model.NewPointer(join.Datetime)
+		channelCopy := channel
```

Comment:
Do you know why a copy is needed?

Reply:
Nope, old code that I didn't notice. Fixed here and below. Thanks!  (It's all going to be combined in a few PRs anyway... Looking forward to that one.)